### PR TITLE
test(swc): Add Bazel tests

### PR DESCRIPTION
**Pull request by realtimetodie**
Adds an integration test that simulates a Bazel environment using symbolic links to sources.

Related

- https://github.com/swc-project/swc/pull/6930

### CODE REVIEW

**Review by kdy1 — DISMISSED** [review]
Thank you so much!

---
swc-bump:
 - dbg-swc

**Comment by kdy1** [comment]
Oh CI failed

**Comment by realtimetodie** [comment]
This can only be merged after #6930.

File: crates/swc/tests/projects.rs [hunk 1]
```diff
@@ -938,6 +943,74 @@ fn json_schema() {
...
+        // We ignore the error silently here as this test will fail for
+        // unprivileged users.
+        #[cfg(target_os = "windows")]
+        {
+            symlink_dir(src_dir.clone(), symlink_build_path.clone()).unwrap_or_else(|_| {
+                if std::env::var("GITHUB_ACTIONS").is_ok() {
+                    panic!("Failed to create symbolic link");
+                };
+
+                return ();
```
Comment (kdy1) [inline crates/swc/tests/projects.rs:979]: Please remove this line

File: crates/swc/tests/projects.rs [hunk 2]
```diff
@@ -938,6 +943,74 @@ fn json_schema() {
...
+        let test_dir = current_dir().unwrap().join("tests/projects/bazel-support");
+
+        let src_dir = test_dir.join("src");
+        let modules_dir = test_dir.join("modules");
+
+        let symlink_build_path = test_dir.join("build");
+        let _ = remove_file(symlink_build_path.clone());
+
+        let symlink_source_path = src_dir.join("modules");
+        let _ = remove_file(symlink_source_path.clone());
```
Comment (kdy1) [inline crates/swc/tests/projects.rs:961]: This clone is redundant

Comment (kdy1) [inline crates/swc/tests/projects.rs:961]:
```suggestion
        let _ = remove_file(&symlink_source_path);
```

File: crates/swc/tests/projects.rs [hunk 3]
```diff
@@ -938,6 +943,74 @@ fn json_schema() {
...
+    testing::run_test2(false, |cm, handler| {
+        let c = swc::Compiler::new(cm.clone());
+
+        let test_dir = current_dir().unwrap().join("tests/projects/bazel-support");
+
+        let src_dir = test_dir.join("src");
+        let modules_dir = test_dir.join("modules");
+
+        let symlink_build_path = test_dir.join("build");
+        let _ = remove_file(symlink_build_path.clone());
```
Comment (kdy1) [inline crates/swc/tests/projects.rs:958]:
```suggestion
        let _ = remove_file(&symlink_build_path);
```

File: crates/swc/tests/projects.rs [hunk 4]
```diff
@@ -938,6 +943,74 @@ fn json_schema() {
...
+
+        let symlink_source_path = src_dir.join("modules");
+        let _ = remove_file(symlink_source_path.clone());
+
+        #[cfg(target_family = "unix")]
+        {
+            symlink(src_dir, symlink_build_path.clone()).unwrap();
```
Comment (kdy1) [inline crates/swc/tests/projects.rs:965]: Same here

File: crates/swc/tests/projects.rs [hunk 5]
```diff
@@ -938,6 +943,74 @@ fn json_schema() {
...
+            symlink(modules_dir, symlink_source_path).unwrap();
```
Comment (kdy1) [inline crates/swc/tests/projects.rs:966]: Same here

File: crates/swc/tests/projects.rs [hunk 6]
```diff
@@ -938,6 +943,74 @@ fn json_schema() {
...
+        }
+
+        // On Windows, creating symbolic links requires debug privileges.
+        // We ignore the error silently here as this test will fail for
+        // unprivileged users.
+        #[cfg(target_os = "windows")]
+        {
+            symlink_dir(src_dir.clone(), symlink_build_path.clone()).unwrap_or_else(|_| {
```
Comment (kdy1) [inline crates/swc/tests/projects.rs:974]: Same here

**Review by kdy1 — CHANGES_REQUESTED** [review]
Honestly, I'm not sure if this should be treated as a valid test case.
Does tsc work in this way? I'm going to patch `jsc.paths` to match one of tsc soon, and I think this test case can interfere it.

**Pull request closed without merging**